tests/e2e/raw_block_volume.go

```diff
@@ -596,4 +614,369 @@ var _ = ginkgo.Describe("raw block volume support", func() {
+		}
+		isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volumeID, vmUUID)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(isDiskAttached).To(gomega.BeTrue(), fmt.Sprintf("Volume is not attached to the node, %s", vmUUID))
+
+		// Get the size of block device and verify if device is accessible by performing write and read.
+		// Write and read some data on raw block volume inside the pod.
+		// Use same devicePath for raw block volume here as used inside podSpec by createPod().
+		// Refer setVolumes() for more information on naming of devicePath.
+		volumeIndex := 1
+		devicePath := fmt.Sprintf("%v%v", pod_devicePathPrefix, volumeIndex)
+		ginkgo.By(fmt.Sprintf("Check size for block device at devicePath %v before expansion", devicePath))
+		originalBlockDevSize, err := getBlockDevSizeInBytes(f, namespace, pod, devicePath)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By(fmt.Sprintf("Write and read data on raw volume attached to: %v at path %v", pod.Name,
+			pod.Spec.Containers[0].VolumeDevices[0].DevicePath))
+		verifyDataOnRawBlockVolume(namespace, pod.Name, devicePath, "This is write before expansion", 0)
+
+		defer func() {
+			// Delete Pod.
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err := fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client,
+				pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(),
+				fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+			if guestCluster {
+				ginkgo.By("Waiting for 30 seconds to allow CnsNodeVMAttachment controller to reconcile resource")
+				time.Sleep(waitTimeForCNSNodeVMAttachmentReconciler)
+				verifyCRDInSupervisorWithWait(ctx, f, pod.Spec.NodeName+"-"+svcPVCName,
+					crdCNSNodeVMAttachment, crdVersion, crdGroup, false)
+			}
+		}()
+
+		ginkgo.By("Increase PVC size and verify online volume resize")
+		increaseSizeOfPvcAttachedToPod(f, client, namespace, pvc, pod)
+
+		ginkgo.By("Wait for block device size to be updated inside pod after expansion")
+		isPvcExpandedInsidePod := false
+		for !isPvcExpandedInsidePod {
+			blockDevSize, err := getBlockDevSizeInBytes(f, namespace, pod, devicePath)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			if blockDevSize > originalBlockDevSize {
+				ginkgo.By("Volume size updated inside pod successfully")
+				isPvcExpandedInsidePod = true
+			} else {
+				ginkgo.By(fmt.Sprintf("updating volume size for %q. Resulting volume size is %d", pvc.Name, blockDevSize))
+				time.Sleep(30 * time.Second)
+			}
+		}
+
+		// Verify original data on raw block volume after expansion
+		// Write data on expanded space to verify the expansion is sucessful and accessible.
```

Comment:
move this comment closer to the execution part

Reply:
done